src/ucm/bistro/bistro.c

```diff
@@ -53,7 +54,13 @@ ucs_status_t ucm_bistro_apply_patch(void *dst, void *patch, size_t len)
         return status;
     }
 
+#if defined(__riscv)
+    __sync_synchronize();
```

Comment:
maybe sync on all platforms? or use memory barriers macros?
(to avoid if defined here)

Reply:
the riscv processor has a `[fence.i](https://five-embeddev.com/riscv-isa-manual/latest/zifencei.html)` instruction synchronizing both the instruction and data caches; it exists for self modifying code (JITs). would you and @shamisp be open to each architecture providing a macro that is selectively implemented to optionally support this processor feature?

Reply:
yes; for now it can be empty for all other architectures

Reply:
I have it implemented for arm, but we had no reason to expose it outside of arch:
https://github.com/openucx/ucx/blob/57615271cb77440ccb131d581e4f108a2b234fbc/src/ucs/arch/aarch64/cpu.h#L39

We have `#define ucs_memory_cpu_fence` and I think this one can follow similar naming  convention `#define ucs_instruction_cpu_fence`

Reply:
I'm quire a bit confused why we need I-berrier here. Eventually, when patched applied we invoke `ucs_clear_cache` which flushes the I-cache.

Reply:
@ct-clmsn Do you still need the `if defined`, even after fixing cache flush ?